File: Data Structures & Algorithms/insert-into-a-binary-search-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode insertIntoBST(TreeNode root, int val) {

        if(root == null) {
            TreeNode node = new TreeNode(val);
            return node;
        }

        // if(root.left == null && val < root.val) {
        //     TreeNode leftNode = new TreeNode(val);
        //     root.left = leftNode;
        // }

        // if(root.right == null && val > root.val) {
        //     TreeNode rightNode = new TreeNode(val);
        //     root.right = rightNode;
        // }

         if(val > root.val) {
            root.right = insertIntoBST(root.right, val);
         } else if (val < root.val) {
            root.left = insertIntoBST(root.left, val);
         }

         return root;
        
    }
}

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

from collections import deque

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:

        if not root:
            return []

        result = []

        queue = deque()
        queue.append(root)
        result.append(root.val)

        while queue:

            size = len(queue)
            temp_arr = []
            for i in range(size):
                node = queue.popleft()

                if node.left:
                    queue.append(node.left)
                    temp_arr.append(node.left.val)

                if node.right:
                    queue.append(node.right)
                    temp_arr.append(node.right.val)
            if len(temp_arr) > 0:
                result.append(temp_arr[len(temp_arr)-1]) 
        return result

            

        
        